### PR TITLE
Lesson 9 Ready for Review

### PR DESCRIPTION
**Pull request by srepking**
Thank you!

### CODE REVIEW

File: students/srepking/Lesson09/donors.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+
+
+class Group:
+
+    def __init__(self):
+
+        self._donor_raw = {}
+
+    def get_donors(self):
```
Comment (natasha-aleksandrova) [inline students/srepking/Lesson09/donors.py]: common practice is to use a decorator for property:
```
@property
def get_donors(self):
```

Comment (natasha-aleksandrova) [reply students/srepking/Lesson09/donors.py]: however.. no need for decorator here if you're not applying any special logic to that attribute.

File: students/srepking/Lesson09/donors.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        return self._donor_raw
+
+    donors = property(get_donors)
+
+    def search(self, donor):
+        if self._donor_raw.get(donor) is None:
+            return
+        else:
+            return donor
```
Comment (natasha-aleksandrova) [inline students/srepking/Lesson09/donors.py]: can be simplified to:
```
def search(self, donor):
    return self._donor_raw.get(donor)
```
it's identical to your code as far as results.

File: students/srepking/Lesson09/donors.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    def add(self, donor, donation):
+        if self.search(donor) is None:
+            self._donor_raw[donor] = [donation]
+        else:
+            self._donor_raw[donor].append(donation)
+
+    def create_list(self):
```
Comment (natasha-aleksandrova) [inline students/srepking/Lesson09/donors.py]: more appropriate name would be -> `def print_donors(self):`

File: students/srepking/Lesson09/donors.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    def column_name_width(donor_summary):
+        name_list = list(donor_summary.keys())  # creates a list of keys
+        name_wi = 11  # Establish minimum column width
+        for i in name_list:
+            if len(i) > name_wi:
+                name_wi = (len(i))  # width of name column
+        return name_wi
+
+    @staticmethod
+    def column_total_width(donor_summary):
```
Comment (natasha-aleksandrova) [inline students/srepking/Lesson09/donors.py]: nice use of staticmethods

File: students/srepking/Lesson09/donors.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+class Individual:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def thank_you(donor, donation):
+        """Add a donation to a donors records and print a report."""
+        return ('Thank you so much for the generous gift of ${0:.2f}, {1}!'
+              .format(float(donation), donor))
```
Comment (natasha-aleksandrova) [inline students/srepking/Lesson09/donors.py]: This class should be utilized in your `Group` class.

The relation ship here is one to many; you have `Donor` and `DonorCollection` - ideally `DonorCollection` would reference many `Donor` objects.

Here is an example of what I mean:
https://repl.it/@naaleksandrova/Lesson9Example

File: students/srepking/Lesson09/donors.py (hunk 2 above)
Comment (srepking) [reply students/srepking/Lesson09/donors.py]: Yes! I fixed it and Thank you!

File: students/srepking/Lesson09/donors.py (hunk 1 above)
Comment (srepking) [reply students/srepking/Lesson09/donors.py]: I looked at your example showing how you made the relationship between single donor objects and the collection. Very nice! I incorporated this approach that you used, and learned from it. Thank you!

I deleted the @property for this def, but used it elsewhere.

File: students/srepking/Lesson09/donors.py (hunk 3 above)
Comment (srepking) [reply students/srepking/Lesson09/donors.py]: Updated my code.

Comment (srepking) [reply students/srepking/Lesson09/donors.py]: I made the change.

File: students/srepking/Lesson09/donors.py (hunk 5 above)
Comment (srepking) [reply students/srepking/Lesson09/donors.py]: In your example, you also have the thank you letter in the Donor Class. Let me know if this comment still stands after my next pull request. I have incorporated your example, and now my DonorCollection is referencing many Donor objects. This is a much nicer way of doing this! Thank you!

File: students/srepking/Lesson09/donors.py (hunk 1 above)
Comment (natasha-aleksandrova) [reply students/srepking/Lesson09/donors.py]: glad it helped!

**Pull request merged**